client/blocks/comments/comment-actions.scss

```diff
@@ -6,13 +6,16 @@
 	margin-top: -6px;
 
 	button {
+		@extend %button;
```

Comment:
Is this extend really needed? I would expect the element get all the needed styles from the `.button` class.

Looking at devtools, I see some differences in, e.g., `font-size` and `padding`. But I'm a bit confused by all that rather than understanding the intent:

<img width="592" alt="Screenshot 2020-09-02 at 14 32 19" src="https://user-images.githubusercontent.com/664258/91985696-cf457100-ed2c-11ea-94a3-9414bb7cea34.png">

Reply:
> Looking at devtools, I see some differences in, e.g., font-size and padding. But I'm a bit confused by all that rather than understanding the intent:

The problem is that you need to also remove the global button styles for this to make sense. If you remove the global styles, there are several issues that arise. Therefore, I added the extends in order to be able to maintain those styles while removing one thing from depending on them globally. Future change sets can continue to extend `%button` until we've fully removed the dependency on the global application of those styles to all buttons.

Right now, you're seeing both the global and the local application of those styles, so it's not going to make a lot of sense (it'll look like duplication mostly). If you remove the global button styles from `_reset.scss` it might make more sense.

On a separate note, I think I should actually open a PR to fix this in the component itself, because it currently relies on the Calypso resets as well and should have those styles built in. If you remove the resets, even the `Button` component will look bad:

<img width="278" alt="Captura de Tela 2020-09-02 às 09 49 47" src="https://user-images.githubusercontent.com/24264157/92012711-a491f300-ed01-11ea-8172-5d1f0aa64814.png">

Reply:
Thank you for pointing this out. I was indeed going about this the wrong way. I've updated the branch with new code that doesn't rely on the global styles.

The problem I was running into was that I was removing the global styles to check how my changes were working, and getting confused with the like button, which was using a different method of getting the `button` tag. I've added something of a hack to make it work using the `Button` component, but I'm not sure if there's a better way to do this. I've never worked with this `tagName` pattern in React before, so I'm kind of just guessing based on what I can imagine will work based on the implementation I am seeing. I would especially love feedback with respect to that.